up-l1/README.adoc

```diff
@@ -122,9 +122,21 @@ On the other hand, unsuccessful completion of this method does not necessarily m
 
 NOTE: The above strategy for retrying failed attempts to send a message results in https://www.cloudcomputingpatterns.org/at_least_once_delivery/[at-least-once delivery]. Recipient(s) of these messages should therefore be https://www.cloudcomputingpatterns.org/idempotent_processor/[Idempotent Processors].
 
-UTransport implementations
 
+[.specitem,oft-sid="des~utransport-send-preserve-data~1",oft-needs="impl,utest"]
+--
 * *MUST* preserve all of the message's meta data and payload during transmission
+--
+
+[.specitem,oft-sid="req~utransport-send-error-invalid-parameter~1",oft-needs="impl,utest"]
+--
+* *MUST* fail with a `UCode.INVALID_ARGUMENT` if the passed UMessage failed validation
+--
+
+[.specitem,oft-sid="req~utransport-send-error-permission-denied~1",oft-needs="impl,utest"]
+--
+* *MUST* fail with a `UCode.PERMISSION_DENIED` if a non-streamer client tries to send a message with `UAttributes.source` that is different than the source address associated with the transport, this is to avoid address spoofing.
```

Comment:
```suggestion
* *MUST* fail with a `UCode.PERMISSION_DENIED` if a non-streamer client tries to send a message with `UAttributes.source` that is different than the source address associated with the transport. Done to avoid address spoofing.
```

How do we propose to activate / deactivate this logic within each transport? If all we do is have an alternative constructor uStreamer constructor, it's possible for someone to simply use that constructor and spoof.

Is there some kind of authentication or verification that should be done? I'm not well-versed in how to allow this for the uStreamer, but prevent for all other users of a transport on a device.

Reply:
FMPOV this taps into the domain of authentication/authorization. A uEntity and/or uStreamer will need to be run in the context of an authenticated identity with some authorities granted by the system owner. This might be as simple as a JWT that has been issued by an identity provider. A uStreamer in this context would then have a specific authority that is being checked by the transport during runtime and which allows the streamer to send messages _on behalf of_ other uEntities.

Reply:
Agree with you @sophokles73 but trying to see how I can word this into the transport spec as we need to ensure that address spoofing cannot happen by clients. I wonder if we bake this into the LocalUriAuthority definitions perhaps?

Reply:
So where did we land on this? Leave open-ended on how to achieve the no-spoofing requirement, give some suggestions, or mandate something?

Reply:
@PLeVasseur If it is OK with you, I would like to add that in a subsequent PR unless you think it needs to be addressed in this PR now.

Reply:
If it's a follow-up, then I'm good.

Reply:
@PLeVasseur I created #228 to track this open item